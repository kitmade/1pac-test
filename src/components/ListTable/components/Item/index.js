import React from "react";
import PropTypes from "prop-types";

function Item(props) {
  const { data } = props.data;
  console.log(props);
  return (
    <tr>
      <td>{props.index}</td>
      {Object.values(data[0]).map((e, idx) => (
        <td key={`item-${props.index}-${idx}`}>{e ? (Array.isArray(e) ? e.join(", ") : e) : ""}</td>
      ))}
    </tr>
  );
}

Item.propTypes = {
  data: PropTypes.shape({
    data: PropTypes.array,
    href: PropTypes.string,
  }),
  index: PropTypes.number,
};
export default Item;
