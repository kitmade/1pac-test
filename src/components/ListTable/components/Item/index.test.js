import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Item from "./";

let defaultData = {
  data: [{ title: "test", date_created: "2000/01/01" }],
  nasa_id: "test_id",
};

describe("Item", () => {
  let data = {};

  beforeEach(() => {
    data = defaultData;
  });

  it("should match snapshot", () => {
    const comp = renderer.create(<Item data={data} />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it("should match snapshot when data is empty", () => {
    data = { data: [{}], nasa_id: "" };
    const comp = renderer.create(<Item data={data} />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it("should delete and undo item", () => {
    const { getByText, container } = render(<Item data={data} />);
    fireEvent.click(getByText("remove"));
    expect(container.querySelector("tr").className).toBe("removed-tr");
    fireEvent.click(getByText("undo"));
    expect(container.querySelector("tr").className).toBe("");
  });

  it("should like and unlike item", () => {
    const { getByText, container } = render(<Item data={data} />);
    const likeCheckboxBlock = container
      .getElementsByClassName("table-other-block")[0]
      .querySelector("div");
    fireEvent.click(likeCheckboxBlock.querySelector("input"));
    expect(getByText("liked")).toBeTruthy();
    fireEvent.click(likeCheckboxBlock.querySelector("input"));
    expect(getByText("like")).toBeTruthy();
  });

  it("should be editable", () => {
    const { getByText, container, getAllByTestId } = render(
      <Item data={data} />
    );
    const onEditInputAction = () => {
      fireEvent.change(getAllByTestId("edit-item-input")[0], {
        target: { value: "test changed" },
      });
    };
    fireEvent.click(getByText("edit"));
    expect(getAllByTestId("edit-item-input")).toBeTruthy();
    onEditInputAction();
    fireEvent.click(getByText("cancel"));
    expect(getByText("test")).toBeTruthy();
    fireEvent.click(getByText("edit"));
    onEditInputAction();
    fireEvent.click(getByText("confirm"));
    expect(getByText("test changed")).toBeTruthy();
  });
});
