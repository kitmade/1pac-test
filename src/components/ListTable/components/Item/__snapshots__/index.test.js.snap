// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<tr
  className=""
>
  <td />
  <td>
    test
  </td>
  <td>
    2000/01/01
  </td>
  <td>
    <div
      className="table-other-block"
    >
      <div>
        <input
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          like
        </label>
      </div>
      <p
        onClick={[Function]}
      >
        remove
      </p>
      <p
        onClick={[Function]}
      >
        edit
      </p>
    </div>
  </td>
</tr>
`;

exports[`Item should match snapshot when data is empty 1`] = `
<tr
  className=""
>
  <td />
  <td>
    <div
      className="table-other-block"
    >
      <div>
        <input
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          like
        </label>
      </div>
      <p
        onClick={[Function]}
      >
        remove
      </p>
      <p
        onClick={[Function]}
      >
        edit
      </p>
    </div>
  </td>
</tr>
`;
